### ISSUE TITLE
import {AsyncResource} from '#async_hooks'; throws Cannot find package "#async_hooks" error

### ISSUE THREAD

**Issue by btakita** [oven-sh/bun#7488]
### What version of Bun is running?

1.0.14+d8be3e51b

### What platform is your computer?

Linux 6.6.3-arch1-1 x86_64 unknown

### What steps can reproduce the bug?

```js
import {AsyncResource} from '#async_hooks';
```

### What is the expected behavior?

`AsyncResource` from `node:async_hooks` library should be loaded.

### What do you see instead?

`Cannot find package "#async_hooks"`

### Additional information

After the Astrojs v4 upgrade, I got the error `Cannot find package "#async_hooks"`. This comes from the [p-limit](https://github.com/sindresorhus/p-limit) library.

In https://github.com/sindresorhus/p-limit/issues/76, I suggested they use:

```js
import {AsyncResource} from 'node:async_hooks';
```

**Comment by TiBianMod** [comment]
Right now, Bun can resolve imports only from files (i think)
> "node": "async_hooks", <--- This is a import from node

```json
{
  "imports": {
    "#async_hooks": {
      "node": "async_hooks",
      "default": "./async-hooks-stub.js"
    }
  }
}
```

> [p-limit package.json](https://github.com/sindresorhus/p-limit/blob/main/package.json#L18:L23)

**Comment by btakita** [comment]
The astro bug was [closed](https://github.com/withastro/astro/issues/9338#issuecomment-1842552258) as "not planned". This means Astro v4 does not work with bun until package.json imports is implemented or if p-limit changes it's implementation.

**Comment by Jarred-Sumner** [comment]
The bug is, most likely, due to us not yet rewriting the imports in `#imports` with the same code path as we do at runtime. It should be a 5 line fix or so and we will fix it today

**Comment by btakita** [comment]
I appreciate it.

**Comment by siinghd** [comment]
Hi there,

I've been working on a project using Bun.js and I ran into an issue. I was using Bun.js 1.0.15 and I updated to 1.0.16. This update solved some problems during development (when I run `bun run app.js`). But now, when I try to build my project with `bun build app.ts --outdir ./out --target bun` or `bun build app.ts --outdir ./out --target node`, I get this error:

```js
error: File not found "async_hooks"
    at async_hooks
```

I'm using `p-limit` in my project. Has anyone else faced this issue? Any suggestions on how to fix it or work around it would be really helpful.

**Comment by Jarred-Sumner** [comment]
> Hi there,
> 
> I've been working on a project using Bun.js and I ran into an issue. I was using Bun.js 1.0.15 and I updated to 1.0.16. This update solved some problems during development (when I run `bun run app.js`). But now, when I try to build my project with `bun build app.ts --outdir ./out --target bun` or `bun build app.ts --outdir ./out --target node`, I get this error:
> 
> ```js
> error: File not found "async_hooks"
>     at async_hooks
> ```
> 
> I'm using `p-limit` in my project. Has anyone else faced this issue? Any suggestions on how to fix it or work around it would be really helpful.

@siinghd I missed the `bun build` case, fixed in 989170ee483419289083a4a2db30ee890480904f
